Makefile

```diff
@@ -0,0 +1,83 @@
+RTEMS_API = 6
+RTEMS_CPU = sparc
+RTEMS_BSP = gr712rc-qual-only
+RTEMS_ROOT = /opt/rtems-6-sparc-gr712rc-smp-3
+
+export SIS_NAME = sis
+export SIS_VERSION = 2.29
+
+export PKG_CONFIG = $(RTEMS_ROOT)/lib/pkgconfig/$(RTEMS_CPU)-rtems$(RTEMS_API)-$(RTEMS_BSP).pc
+export SIS_DIR = sis
+export SRC_DIR = src
+export BUILD_DIR = build
+
+SRC = $(wildcard $(SRC_DIR)/*/*.c)
+INCL = $(SRC_DIR)/ $(sort $(dir $(wildcard $(SRC_DIR)/*/*.h)))
+PROJ_OBJS = $(addprefix $(BUILD_DIR)/,$(patsubst %.c,%.o,$(SRC)))
+
+DEPFLAGS = -MT $@ -MD -MP -MF $(basename $@).d
+WARNFLAGS = -Wall -Wextra
+OPTFLAGS = -Os -ffunction-sections -fdata-sections
+ABI_FLAGS = $(shell pkg-config --cflags $(PKG_CONFIG))
+EXEEXT = .exe
+
+LDFLAGS = $(shell pkg-config --libs $(PKG_CONFIG))
+CFLAGS = -g $(DEPFLAGS) $(WARNFLAGS) $(ABI_FLAGS) $(OPTFLAGS) $(addprefix -I,$(INCL)) -DRTEMS_API_$(RTEMS_API) -DRTEMS_SIS
+
+CCLINK = $(CC) $(CFLAGS) -Wl,-Map,$(basename $@).map
+
+export PATH := $(RTEMS_ROOT)/bin:$(PATH)
+export AR = $(RTEMS_ROOT)/sparc-rtems6/bin/ar
+export AS = $(RTEMS_ROOT)/sparc-rtems6/bin/as
+export CC = $(RTEMS_CPU)-rtems$(RTEMS_API)-gcc
+export LD = $(RTEMS_ROOT)/sparc-rtems6/bin/ld
+export NM = $(RTEMS_ROOT)/sparc-rtems6/bin/nm
+export OBJCOPY = $(RTEMS_ROOT)/sparc-rtems6/bin/objcopy
+export RANLIB = $(RTEMS_ROOT)/sparc-rtems6/bin/ranlib
+export SIZE = $(RTEMS_CPU)-rtems$(RTEMS_API)-size
+export STRIP = $(RTEMS_ROOT)/sparc-rtems6/bin/strip
+
+DEMO = $(BUILD_DIR)/demo
+DEMO_DIR = ./demo
+DEMO_OBJ = $(DEMO_DIR)/main.o
+
+TEST = $(BUILD_DIR)/test
+TEST_DIR = ./test
+TESTS = $(TEST_DIR)/main.c $(wildcard $(TEST_DIR)/*/*/*.c)
+TEST_INCL = $(TEST_DIR)/ $(sort $(dir $(wildcard $(TEST_DIR)/*/*/*.h)))
+TEST_OBJS = $(patsubst %.c,%.o,$(TESTS))
+
+all: $(BUILD_DIR)
+
+$(BUILD_DIR):
+	mkdir -p $(BUILD_DIR)/$(SIS_DIR)
```

Comment:
Please reorganize makefiles so that internal Makefiles are independent from the top level makefile, and builds of artifacts are contained in respective makefiles (libraries (.a) can be sourced from other makefiles as dependencies)